src/Math-ODE/PMAB2Stepper.class.st

```diff
@@ -15,9 +15,10 @@ PMAB2Stepper class >> order [
 
 { #category : #stepping }
 PMAB2Stepper >> doStep: aState prevState: prevState time: t [
-	self stepSize isNil
-		ifTrue: [ self error: 'step size required by stepper' ].
-	^ (self stepSize / 2) * (3 * (system state: aState time: t) - (system state: prevState time: t - self stepSize)) + aState
+
+	self stepSize ifNil: [ self error: 'step size required by stepper' ].
+	^ self stepSize / 2 * (3 * (system state: aState time: t)
```

Comment:
It looks like this computation may represent the same concept, but each class performs it differently. Is that right? At the very least, togive it a name,this computation might be extracted to an intention-revealing method or a local variable.

Reply:
Also, I think you might be able to refactor these methods towards the template design pattern, and then you might be led to this method being pulled up to the parent class, removing much more duplication, and subclasses providing behaviour to the parent class. I think this may simplify the code significantly. Thus I would encourage you to explore/introduce that in this PR or in the next.